### ISSUE TITLE
Master branch

### ISSUE THREAD

**Issue by kabalin** [MaharaProject/moodle-assignsubmission_mahara#16]
Since there are no major version-specific changes in the plugin code between 2.7, 2.8 and 2.9, I suggest to start using master branch as a main plugin branch and use independent plugin versioning. There is no need to differentiate versions of the same codebase between branches, in fact it may introduce issues when same upgrade code is applied twice under different version numbers.

**Comment by agwells** [comment]
Fair enough. Another advantage to reducing them to one branch, is that I don't usually have a lot of time available for this plugin, so making things simpler is probably better.

And if we do need to add some version-specific code at some point in the future, we can always adopt a new separate branch again. Or surround it in a construct like "if ($cfg->version > XXXX)".

I'll take a look at how the Moodle DB handles updates these days, and decide whether or not we still need to have separate "releases" for the different Moodle versions (even if they contain identical code).

**Comment by kabalin** [comment]
> And if we do need to add some version-specific code at some point in the future, we can always adopt a new separate branch again.

Yep, that what I thought, for version-specific patches.

>I'll take a look at how the Moodle plugin site handles updates these days, and decide whether or not we still need to have separate "releases" for the different Moodle versions (even if they contain identical code).

I think these days you just tag release with versions it supposed to be compatible with, at least for [JWPlayer](https://moodle.org/plugins/view/filter_jwplayer) I was using just one branch (will be using two since next release because JW6 and JW7 have significant differences).

**Comment by kabalin** [comment]
@agwells Shall I push "master" to moodle-assignsubmission_mahara, so that you would review it here and relase when ready?

**Comment by agwells** [comment]
I've just pushed it up and renamed the moodle27 and moodle28 branches to indicate they're deprecated.

The only thing I'm hesitant about, is saying it supports "Moodle 2.7+", because someone could come across that a year from now and think it means that we've tested it against Moodle 3.1 when we have not. I'd rather put a ceiling on that to indicate which versions we know it works with "Moodle 2.7 - 2.9". And shift that as each new release comes out and we get some sense of whether it works with the new release or not.
